examples/evaluation/evaluate_with_custom_code_metrics.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import os
+
+import openai
+import pandas as pd
+
+import mlflow
+from mlflow.metrics import make_metric
+from mlflow.metrics.base import MetricValue
+from mlflow.metrics.metric_definitions import standard_aggregations
+
+logging.getLogger("mlflow").setLevel(logging.ERROR)
+
+
+# Uncomment the following lines to run this script without using a real OpenAI API key.
+# os.environ["OPENAI_API_KEY"] = "test"
+
+assert "OPENAI_API_KEY" in os.environ, "Please set the OPENAI_API_KEY environment variable."
+
+
+# Helper function to check if a string is valid python code
+def is_valid_python_code(code: str) -> bool:
+    try:
+        compile(code, "<string>", "exec")
+        return True
+    except SyntaxError:
+        return False
+
+
+# Create an evaluation function that iterates through the predictions
+def eval_fn(eval_df, metrics):
+    y_pred = eval_df["prediction"]
+    predictions = y_pred.squeeze() if isinstance(y_pred, pd.DataFrame) else y_pred
+
+    scores = [int(is_valid_python_code(prediction)) for prediction in predictions]
+    return MetricValue(
+        scores=scores,
+        aggregate_results=standard_aggregations(scores),
+    )
+
+
+# Create an EvaluationMetric object for the python code metric
+valid_code_metric = make_metric(
+    eval_fn=eval_fn, greater_is_better=False, name="valid_python_code", version="v1"
+)
+
+eval_df = pd.DataFrame(
+    {
+        "input": [
+            "SELECT * FROM ",
+            "import pandas",
+            "def hello_world",
+        ],
+    }
+)
+
+with mlflow.start_run() as run:
+    system_prompt = (
+        "Generate code that is less than 50 characters. Return only python code and nothing else."
+    )
+    logged_model = mlflow.openai.log_model(
+        model="gpt-3.5-turbo",
+        task=openai.ChatCompletion,
+        artifact_path="model",
+        messages=[
+            {"role": "system", "content": system_prompt},
+            {"role": "user", "content": "{question}"},
+        ],
+    )
+
+    results = mlflow.evaluate(
+        logged_model.model_uri,
+        eval_df,
+        model_type="text",
+        custom_metrics=[valid_code_metric],
+    )
+    print(results)
+
+    eval_table = mlflow.load_table("eval_results_table.json", run_ids=[run.info.run_id])
+    print(eval_table)
```

Comment:
Now we can update this code to be:

eval_table = results.table['eval_results_table']